src/sentry/integrations/gitlab/integration.py

```diff
@@ -162,11 +162,12 @@ def get_commit_context(
         except ApiError as e:
             raise e
 
+        date_format_expected = "%Y-%m-%dT%H:%M:%S.%f%z"
```

Comment:
For anyone looking at this PR. The difference is swapping the hardcoded timezone `Z` for `%z`.
```diff
-        date_format_expected = "%Y-%m-%dT%H:%M:%S.%f%z"
+        date_format_expected = "%Y-%m-%dT%H:%M:%S.%fZ"
```